### PR TITLE
[porting][linux] Add support for BLE Mesh app on linux platform

### PR DESCRIPTION
**Pull request by mehulhirpara**
Following commits includes:
1. Fixes for build errors which I encountered while building BLE Mesh app on Linux platform
2. Fixes for functionality failure while running the BLE Mesh app on Linux platform

### CODE REVIEW

File: nimble/host/mesh/src/testing.c [hunk 1]
```diff
@@ -97,7 +97,7 @@ void bt_test_print_credentials(void)
 	struct bt_mesh_subnet *sub;
 	struct bt_mesh_app_key *app_key;
 
-	console_printf("IV Index: %08lx\n", bt_mesh.iv_index);
+	console_printf("IV Index: %08lx\n", (long) bt_mesh.iv_index);
```
[the remark this answers is not in the document]
Comment (mehulhirpara) [reply nimble/host/mesh/src/testing.c]: I am getting build warnings on this and if I am enabling -werror then it is being considered as error. I thought of covering it.

File: nimble/host/mesh/src/testing.c [hunk 2]
```diff
@@ -157,7 +157,11 @@ void bt_test_print_credentials(void)
 
 int bt_test_shell_init(void)
 {
-	return cmd_mesh_init(0, NULL);
+    int err = 0;
+#if MYNEWT_VAL(BLE_MESH_SHELL)
+	err = cmd_mesh_init(0, NULL);
```
[the remark this answers is not in the document]
Comment (mehulhirpara) [reply nimble/host/mesh/src/testing.c]: Sure, I'll added -ENOTSUP

File: porting/npl/linux/src/os_eventq.cc [hunk 3]
```diff
@@ -46,6 +46,17 @@ ble_npl_eventq_init(struct ble_npl_eventq *evq)
     evq->q = new wqueue_t();
 }
 
+bool
+ble_npl_eventq_is_empty(struct ble_npl_eventq *evq)
+{
+    wqueue_t *q = static_cast<wqueue_t *>(evq->q);
+
+    if (q->size())
```
[the remark this answers is not in the document]
Comment (mehulhirpara) [reply porting/npl/linux/src/os_eventq.cc]: I'll add { }

File: porting/npl/linux/src/os_eventq.cc [hunk 4]
```diff
@@ -82,8 +82,10 @@ struct ble_npl_event *ble_npl_eventq_get(struct ble_npl_eventq *evq,
     struct ble_npl_event *ev;
     wqueue_t *q = static_cast<wqueue_t *>(evq->q);
 
-    ev = q->get();
-    ev->ev_queued = 0;
+    ev = q->get(tmo);
+
+    if (ev)
```
[the remark this answers is not in the document]
Comment (mehulhirpara) [reply porting/npl/linux/src/os_eventq.cc]: I'll add { }

File: nimble/host/mesh/src/testing.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (rymanluk) [reply nimble/host/mesh/src/testing.c]: I guess you are getting issue when building on 64bit machine. Ok then, let's just make it a separate patch as this is not related to BLE Mesh Shell disable

[the remark this answers is not in the document]
Comment (mehulhirpara) [reply nimble/host/mesh/src/testing.c]: yeah, I am building it on 64bit machine. I'll create separate commit for this change and remove it form here.

**Comment by mehulhirpara** [comment]
Thanks for review and approval.

**Pull request merged**